logger/structured_logger.go

```diff
@@ -0,0 +1,143 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build structuredlogging
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package logger
+
+import (
+	"context"
+	"io"
+	"log"
+	"log/slog"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type StructuredLog struct {
+	log   *slog.Logger
+	debug bool
+	quiet bool
+	flags int
+}
+
+const (
+	LevelTrace  = slog.Level(-8)
+	LevelNotice = slog.Level(2)
+)
+
+var LevelNames = map[slog.Leveler]string{
```

Comment:
Do we need to use `Leveler` here? I thoguh that plain `Level` is sufficient.
```suggestion
var LevelNames = map[slog.Level]string{
```